### PR TITLE
feat: Use encrypted_ column prefix to mark SQL detections encrypted

### LINKED ISSUE #283: Allow `encrypted_` prefix to be marked as encrypted in SQL detections
[`detect_encrypted_ruby_class_properties`](https://github.com/Bearer/curio/blob/fa16564adaea69ab8b8f658dc83bce6f793be150/pkg/commands/process/settings/custom_detector.yml#L214-L225)  is the verifier we use to mark a SQL detection as `encrypted`.

We could add this new processor [here](https://github.com/Bearer/curio/blob/96cb240f7cf6fc0def8447d9faa61a90222fc98c/pkg/report/output/dataflow/datatypes/get_extras.go#L187-L188) that would automatically flag the fields that are prefixed by `encrypted_` as encrypted.

```rego
package bearer.db_encrypted

 import future.keywords

 default encrypted := false

 encrypted = true {
     startswith(lower(input.target.value.field_name),  "encrypted_")
 }

 verified_by := [
     {
         "detector": "db_encrypted",
     }

```

### PR DESCRIPTION
**Pull request by spdawson**
## Description

Closes #283

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [X] I've updated or added documentation if required.
- [X] I've included usage information in the description if CLI behavior was updated or added.
- [X] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/commands/process/settings/processors/db_encrypted.rego [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package bearer.db_encrypted
+
+import future.keywords
+
+encrypted contains detection if {
+	detection := input.target_detections[_]
```
Comment (cfabianski) [inline pkg/commands/process/settings/processors/db_encrypted.rego:14]: Should we limit it to `SQL` (includes `create_table_sql` custom detection) or `schema_rb`?

File: integration/flags/.snapshots/TestReportFlags-report-dataflow-db-encrypted [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                    )
+    - name: Gender
+      detectors:
+        - name: detect_sql_create_public_table
+          locations:
+            - filename: integration/flags/testdata/db_encrypted/structure.sql
+              line_number: 9
+              encrypted: true
+              verified_by:
+                - detector: detect_sql_create_public_table
```
Comment (cfabianski) [inline integration/flags/.snapshots/TestReportFlags-report-dataflow-db-encrypted]: This doesn't seem correct

File: pkg/commands/process/settings/processors/db_encrypted.rego [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	detection.detector_type in ["schema_rb", "detect_sql_create_public_table"]
+
+	verifications := [verification |
+		encrypted_detection := sql_encrypted[_]
+		detection.value.object_name == encrypted_detection.value.object_name
+		detection.value.field_name == encrypted_detection.value.field_name
+
+		verification := {
+			"detector": encrypted_detection.detector_type,
```
Comment (cfabianski) [inline pkg/commands/process/settings/processors/db_encrypted.rego]: That's probably why the detector reported in the verified_by isn't correct :)

File: pkg/commands/process/settings/processors/db_encrypted.rego [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+			"filename": encrypted_detection.source.filename,
+			"line_number": encrypted_detection.source.line_number,
```
Comment (cfabianski) [inline pkg/commands/process/settings/processors/db_encrypted.rego]: Don't think that's required but happy to leave it if you think that's useful

**Review by cfabianski — COMMENTED** [review]
Couple of comments but looks very close

File: pkg/commands/process/settings/processors/db_encrypted.rego (hunk 4 above)
Comment (spdawson) [reply pkg/commands/process/settings/processors/db_encrypted.rego]: Will get rid of this

**Review by cfabianski — APPROVED** [review]

**Pull request merged**